multiqc/modules/bcl2fastq/bcl2fastq.py

```diff
@@ -152,6 +153,7 @@ def parse_file_as_json(self, myfile):
                     run_data[lane]["qscore_sum"] += readMetric["QualityScoreSum"]
                     run_data[lane]["samples"][sample]["yieldQ30"] += readMetric["YieldQ30"]
                     run_data[lane]["samples"][sample]["qscore_sum"] += readMetric["QualityScoreSum"]
+                    run_data[lane]["samples"][sample]["trimmed_bases"] += readMetric["TrimmedBases"]
             undeterminedYieldQ30 = 0
             undeterminedQscoreSum = 0
             if "Undetermined" in conversionResult:
```

Comment:
You missed adding the `TrimmedBases` stats from the `Undetermined`. So when the later code gets to this "sample" it crashes with a `KeyError`.